starky/src/proof.rs

```diff
@@ -56,4 +89,34 @@ impl<F: RichField + Extendable<D>, const D: usize> StarkOpeningSet<F, D> {
             quotient_polys: eval_commitment(zeta, quotient_commitment),
         }
     }
+
+    // Note: Can't implement this directly on `Challenger` as it's in a different crate.
+    pub fn observe<H: Hasher<F>>(&self, challenger: &mut Challenger<F, H>) {
```

Comment:
This is only somewhat related to your PR, so maybe it should be a separate change, but it seems like we could replace `Challenger`'s `observe_opening_set` method with one that takes `FriOpenings`. Then methods like this could just become

```rust
challenger.observe_fri_openings(self.to_fri_openings());
```

(It would change the order that openings are observed in; not sure if any other code would need updating to reflect that.)